lib/rex/ui/text/shell.rb

```diff
@@ -57,8 +57,8 @@ def initialize(prompt, prompt_char = '>', histfile = nil, framework = nil, name
     self.hist_last_saved = 0
 
     # Static prompt variables
-    self.local_hostname = ENV['HOSTNAME'] || `hostname`.split('.')[0] || ENV['COMPUTERNAME']
-    self.local_username = ENV['USER'] || `whoami` || ENV['USERNAME']
+    self.local_hostname = ENV['HOSTNAME'] || try_exec('hostname')&.split('.')&.first&.rstrip || ENV['COMPUTERNAME'] || '???'
+    self.local_username = ENV['USER'] || try_exec('whoami')&.rstrip || ENV['USERNAME'] || '???'
```

Comment:
If these is left as `nil` instead of `???`:

```suggestion
    self.local_hostname = ENV['HOSTNAME'] || try_exec('hostname')&.split('.')&.first&.rstrip || ENV['COMPUTERNAME']
    self.local_username = ENV['USER'] || try_exec('whoami')&.rstrip || ENV['USERNAME']
```

It looks like it should get correctly handled elsewhere in the system:

https://github.com/rapid7/metasploit-framework/blob/040636d4f4f565004651ed53de01558c79d9b13a/lib/rex/ui/text/shell.rb#L459-L460

https://github.com/rapid7/metasploit-framework/blob/040636d4f4f565004651ed53de01558c79d9b13a/lib/rex/ui/text/shell.rb#L463-L464